core/client/fs/src/main/java/alluxio/client/file/FileInStream.java

```diff
@@ -623,28 +629,30 @@ private void seekInternalWithCachingPartiallyReadBlock(long pos) throws IOExcept
     } else {
       mPos = pos / mBlockSize * mBlockSize;
       updateStreams();
-      readCurrentBlockToPos(pos);
+      if (mCurrentCacheStream != null) {
+        cacheCurrentBlockToPos(pos);
+      } else if (mCurrentBlockInStream != null) {
+        seekInternal(pos);
+      } else {
+        Preconditions.checkState(remaining() == 0);
+      }
```

Comment:
The logic here is a bit convoluted, could you simplify it based on the conditions (ie. mPos does not need to be set to the beginning of the block if we do not plan to cache it)?